recipes/libcufft/meta.yaml

```
@@ -0,0 +1,110 @@
+{% set name = "cufft" %}
+{% set version = "11.0.0.21" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/libcufft/{{ platform }}/libcufft-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: 4650f79fcf377af410afedb56445f1332e8f213ec3c94d660f9a91328c3ecb4a  # [linux64]
+  sha256: e711671995e9d92d16015a4cd3751ec30710a49fa22f2b88a4a42905c248d861  # [ppc64le]
+  sha256: 91aa1ba5e68ca65778222c9a427a9214c4550d2f45e978e51779a207bafc17a3  # [aarch64]
+  sha256: a33f577d48ef432d5cfa8d0ebfa789cc5261af5dd84fd9c608235edc3f769e43  # [win]
+
+
+build:
+  number: 0
+  skip: true  # [osx]
+  run_exports:          # [linux]
+    strong:             # [linux]
+      - __glibc >=2.17  # [linux]
+  missing_dso_whitelist:
+     - /lib64/libdl.so.2            # [linux]
+     - /lib64/libm.so.6             # [linux]
+     - /lib64/libpthread.so.0       # [linux]
+     - /lib64/librt.so.1            # [linux]
+     - /lib64/libgcc_s.so.1         # [linux]
+     - /lib64/libc.so.6             # [linux]
+     - /lib64/ld-linux-x86-64.so.2  # [linux]
+
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+  host:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+  run:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+
+
+outputs:
+  - name: libcufft
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcufft.so.{{ version }}      # [linux]
+        - if not exist %LIBRARY_BIN%\cufft64_11.dll exit 1  # [win]
+    files:
+      - lib/libcufft*.so.*    # [linux]
+      - targets/*/lib/*.so.*  # [linux]
+      - Library\bin           # [win]
+
+  - name: libcufft-dev
+    test:
+      commands:
+        - test -f $PREFIX/include/cufft.h               # [linux]
+        - if not exist %LIBRARY_LIB%\cufft.lib exit 1  # [win]
+    requirements:
+      run:
+        - libcufft >={{ version }}
+        # - cuda-cudart-dev >=12.0
```

Similar to `cuda-cudart` ( https://github.com/conda-forge/staged-recipes/pull/21723#discussion_r1084611245 ), would depend on the static library here

```suggestion
        - libcufft-static >={{ version }}
```

---

Can we not make getting the static libraries more explicit? I.E. conda-forge defaults to using shared libraries, why not have the `dev` package depend on the static library and have someone explicitly request the static library if desired?

---

So the library users of packages would get would be `libcufft`, which just contains the dynamic library. Builds would use `libcufft-dev`, which includes headers, dynamic/static libraries, etc.

That said, if we would rather this be `run_constrained` instead of `run`, that seems fine

---

So `run_constrained` would just ensure we don't have an outdated static library around?  That makes sense to me, but I don't have enough experience to have a real opinion :).

---

Yeah it means if the user installs `libcufft-static`, that requirement is applied. So in this case would be a minimum version

---

I think a previous comment from @robertmaynard on the `cudart` recipe said that getting a `-dev` package that doesn't supply everything needed for building is funky. I think it should have a `run` dependency on either the dynamic or static library.

I understand `cudart` might be special, but for other libraries shouldn't we default to the shared libraries? Static linking in conda packages is typically the exception where I think the space overhead of getting both the shared and static libraries is a reasonable tradeoff.

---

Ok have updated the suggestion above to use `run_constrained`

---

Yes that is what I said. To me it is very weird (broken) that a development package doesn't contain the all the components needed for development( headers, static lib, shared lib ).

---

So that would mean:
- dynamic lib has system runtime dependencies
- static has system runtime dependencies
- dev depends on both dynamic and static (should that actually use `=={{ version.split(',')[0] }}` or exact version matching)?

(I am not clear that the static subpackage can be used without dev, so nobody will ever use them without also having the dynamic libs around.)

Next question would be if:
```
  run_exports:                # [linux]
     strong:                   # [linux]
       - __glibc >=2.17        # [linux]
```
should, at least in principle, be part of the *static* library as:
```
- name: libcufft-static
  run_exports:                # [linux]
     strong:                  # [linux]
       - libcufft-static      # [linux]
```
since arguably if I build depend on `libcufft-dev` and `libcufft-static` (without runtime depend on `libcufft`) I am planning to link in the static lib and require the same runtime exports (sysroot+glibc).

(Sorry, I spend way too much time trying to slowly building a mental model that seems half logical :/)

---

> since arguably if I build depend on libcufft-dev and libcufft-static (without runtime depend on libcufft) I am planning to link in the static lib and require the same runtime exports (sysroot+glibc).

I agree with that.

I don't know enough about the conda rule model to state if the above is correctly maps to our shared mental model